syntax = "proto3";

package datacloud.kafka.console.v1;

import "datacloud/v1/paging.proto";
import "datacloud/kafka/console/v1/cloud.proto";

option go_package = "a.yandex-team.ru/cloud/mdb/datacloud/private_api/datacloud/kafka/console/v1;consolev1";
option java_package = "doublecloud.priv.kafka.console.v1";
option java_outer_classname = "DKCCLS";

service CloudService {
    // Returns list of available clouds.
    rpc List (ListCloudsRequest) returns (ListCloudsResponse);
}

message ListCloudsRequest {
    // Paging information of the request
    datacloud.v1.Paging paging = 1;
}

message ListCloudsResponse {
    // Requested list of available Clouds.
    repeated Cloud clouds = 1;

    // Pagination information of the response
    datacloud.v1.NextPage next_page = 2;
}
